core/math/triangle_mesh.cpp

```diff
@@ -182,7 +182,11 @@ void TriangleMesh::create(const Vector<Vector3> &p_faces, const Vector<int32_t>
 	valid = true;
 }
 
-bool TriangleMesh::intersect_segment(const Vector3 &p_begin, const Vector3 &p_end, Vector3 &r_point, Vector3 &r_normal, int32_t *r_surf_index) const {
+bool TriangleMesh::intersect_segment(const Vector3 &p_begin, const Vector3 &p_end, Vector3 &r_point, Vector3 &r_normal, int32_t *r_surf_index, int *r_face_index) const {
```

Comment:
`int` should probably be a fixed type, `int32_t` would probably be fine, `int64_t` if you want to go crazy.

(It can also be an idea to use `uint32_t` if there was never a need for negative in core code, but I'm not 100% sure the binder works with this.)